src/Components/Web.JS/src/Boot.Server.ts

```diff
@@ -102,10 +102,33 @@ async function initializeConnection(options: BlazorOptions, logger: Logger, circ
     renderQueue.processBatch(batchId, batchData, connection);
   });
 
+  const errorUiReloads = document.querySelectorAll<HTMLElement>("#error-ui .reload");
+  errorUiReloads.forEach(reload => {
+    reload.onclick = function (e) {
+      location.reload();
+      e.preventDefault();
+    };
+  });
+
+  let errorUiDismiss = document.querySelectorAll<HTMLElement>("#error-ui .dismiss");
+  errorUiDismiss.forEach(dismiss => {
+    dismiss.onclick = function (e) {
+      const errorUi = document.querySelector<HTMLElement>("#error-ui");
+      if (errorUi) {
+        errorUi.style.display = 'none';
+      }
+      e.preventDefault();
+    };
+  });
+
   connection.onclose(error => !renderingFailed && options.reconnectionHandler!.onConnectionDown(options.reconnectionOptions, error));
   connection.on('JS.Error', error => {
     renderingFailed = true;
     unhandledError(connection, error, logger);
+    let errorUi = document.querySelector("#error-ui") as HTMLElement;
+    if (errorUi) {
+      errorUi.style.display = 'block';
+    }
   });
```

Comment:
The code here all looks good, but could it be factored out into a separate file? It could export just a single `showErrorNotification` function that we call from inside `connection.on('JS.Error', ...)`.

It would be good if the setup logic you have here between lines 105 and 122 was also inside the `showErrorNotification` function, with some guard to check we only run it on the first invocation. Just so we don't have to run extra DOM-traversing logic during initialization - we can run it lazily instead.

Finally, minor nitpick: in TypeScript we're using `'` rather than `"` quotes.